mmdet/models/anchor_heads/ssd_head.py

```diff
@@ -21,78 +21,115 @@ def __init__(self,
                  anchor_strides=(8, 16, 32, 64, 100, 300),
                  basesize_ratio_range=(0.1, 0.9),
                  anchor_ratios=([2], [2, 3], [2, 3], [2, 3], [2], [2]),
+                 anchor_heights=[],
+                 anchor_widths=[],
                  target_means=(.0, .0, .0, .0),
-                 target_stds=(1.0, 1.0, 1.0, 1.0)):
+                 target_stds=(1.0, 1.0, 1.0, 1.0),
+                 loss_balancing=False,
+                 depthwise_heads=False):
         super(AnchorHead, self).__init__()
         self.input_size = input_size
         self.num_classes = num_classes
         self.in_channels = in_channels
         self.cls_out_channels = num_classes
-        num_anchors = [len(ratios) * 2 + 2 for ratios in anchor_ratios]
+        if len(anchor_heights):
+            assert len(anchor_heights) == len(anchor_widths)
+            num_anchors = [len(anc_conf) for anc_conf in anchor_heights]
+        else:
+            num_anchors = [len(ratios) * 2 + 2 for ratios in anchor_ratios]
         reg_convs = []
         cls_convs = []
         for i in range(len(in_channels)):
-            reg_convs.append(
-                nn.Conv2d(
+            if depthwise_heads:
+                reg_conv = nn.Sequential(
+                    nn.Conv2d(in_channels[i], in_channels[i],
+                              kernel_size=3, padding=1, groups=in_channels[i]),
+                    nn.BatchNorm2d(in_channels[i]),
```

Comment:
BN is commonly not used without multi-gpu synchronization. Are you sure about this setting? Or is there any difference without the bns?

Reply:
I didn't tested it without bns. Lite heads are useful in case of small detectors when batch size is sufficiently large. I had no problems with this head in distributed training mode even without synchronization, because BN has enough samples to learn statistics.
In backbone I also have BNs and everything is ok.